include/circt/Dialect/FIRRTL/FIRRTLDeclarations.td

```diff
@@ -36,10 +36,49 @@ class ReferableDeclOp<string mnemonic, list<Trait> traits = []> :
 /// located in a hardware-creating context, such as the body of a module.
 class HardwareDeclOp<string mnemonic, list <Trait> traits = []> :
   ReferableDeclOp<mnemonic, traits # [
-    ParentOneOf<[
+    ParentOneOf<[ 
       "firrtl::FModuleOp", "firrtl::LayerBlockOp",
       "firrtl::WhenOp", "firrtl::MatchOp", "sv::IfDefOp"]>]> {}
 
+def FormalOp : ReferableDeclOp<"formal", [
```

Comment:
Do we need ReferableDeclOp? I think FormalOp is one of top-level ops so InnerSymbol interface looks a bit weird to me. 

```
/// A declaration that can be referred to by it's name/identity.
class ReferableDeclOp<string mnemonic, list<Trait> traits = []> :
  FIRRTLOp<mnemonic, traits # [HasCustomSSAName,
                               DeclareOpInterfaceMethods<InnerSymbol,["getTargetResultIndex"]>,
                               FNamableOp]> {
```